### PR TITLE
Utilize ProtocolException for HTTP/2 in HttpStress

### LINKED ISSUE #72015: HTTP/2 stress tests failing massively
[Example log](https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/build/builds/1873683/logs/12)

They are reporting a `ProtocolException` with `INTERNAL_ERROR`. From a quick look it looks like the first failure appeared on June 30, after merging #71345.

### PR DESCRIPTION
**Pull request by antonfirsov**
Fixes #72015

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs [hunk 1]
```diff
@@ -303,21 +303,16 @@ public static class ClientOperations
...
-                                        (e.InnerException?.Message?.Contains("CANCEL") ?? false)) // UseHttpSys
-                                    {
-                                        return;
-                                    }
-                                    break;
+                                return;
                             }
                         }
 
+                        // TODO: change when HttpProtocolException gets implemented for HTTP/3
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs:315]: @rkzikm shall we address this:
1. As part of #71432
2. In this PR after merging #71432
3. In a completely separate PR
?

Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs:315]: @rzikm
1. As part of #71432
2. In this PR after merging #71432
3. In a completely separate PR
?

Comment (rzikm) [reply src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs:315]: I will address it in #72095

**Review by rzikm — APPROVED** [review]
LGTM

**Pull request merged**